### PR TITLE
ARC: add XIP support with MWDT toolchain with additional nsim configurations to test it

### PR DESCRIPTION
**Pull request by kokas-a**
Add XIP support with MWDT toolchain. To have it proper tested add separate nsim platforms for XIP (flash + sram) and non-XIP (sram) memory organization in addition to existing nsim_hs platfor with CCMs (ICCM + DCCM) memory organization.

This PR also enables MPU for all nsim hs3x based platforms (like we previously enabled it for qemu_arc_hs) to have proper memory regions permissions.

### CODE REVIEW

**Comment by kokas-a** [comment]
@evgeniy-paltsev , could you please make a review again? 
I closed previous PR(#51810) as outdated.

File: boards/arc/nsim/nsim_hs_sram.dts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2022, Synopsys, Inc. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include "nsim.dtsi"
+#include "nsim-flash-sram-mem.dtsi"
```
Comment (evgeniy-paltsev) [inline boards/arc/nsim/nsim_hs_sram.dts:10]: As we don't use flash in this configuration:

```suggestion
#include "nsim-flat-mem.dtsi"
```

PS: probably we'll rename `nsim-flat-mem.dtsi` to `nsim-sram-mem.dtsi` for less cryptic name :)

File: soc/arc/snps_nsim/linker.ld [hunk 2]
```diff
@@ -33,4 +33,14 @@
 #define SRAM_SIZE		DT_REG_SIZE(DT_CHOSEN(zephyr_sram))
 #endif
 
+/*
+ * flash base address and size
+ */
```
Comment (evgeniy-paltsev) [inline soc/arc/snps_nsim/linker.ld:38]:
```suggestion
/* Flash memory base address and size */
```

**Review by evgeniy-paltsev — CHANGES_REQUESTED** [review]
LGTM, few minor comments.

File: soc/arc/snps_nsim/linker.ld (hunk 2 above)
Comment (kokas-a) [reply soc/arc/snps_nsim/linker.ld]: Done

**Comment by evgeniy-paltsev** [comment]
@kokas-a - just FYI - I've slightly rewrite PR header and description for better clarity.

File: soc/arc/snps_nsim/Kconfig.soc [hunk 3]
```diff
@@ -4,3 +4,4 @@
 config SOC_NSIM
 	bool "Synopsys nSIM simulator for ARC cores"
 	select ARC
+	select CPU_HAS_MPU
```
Comment (evgeniy-paltsev) [inline soc/arc/snps_nsim/Kconfig.soc:7]: This is not in the right place - we shouldn't select it for all nsim SoCs. Let's move it to `soc/arc/snps_nsim/Kconfig`

```
config SOC_NSIM_HS
	bool "Synopsys ARC HS in nSIM"
	select CPU_HAS_FPU
+	select CPU_HAS_MPU
```

**Review by evgeniy-paltsev — CHANGES_REQUESTED** [review]

**Review by evgeniy-paltsev — APPROVED** [review]
I see few issues with MPU protection faults on nsim_hs_flash_xip / nsim_hs_sram in comparison to nsim_hs config, but we can fix them latter as a follow-up.

Let's merge it for now.

**Review by ruuddw — APPROVED** [review]

**Review by abrodkin — APPROVED** [review]

**Pull request merged**